index.js

```diff
@@ -274,10 +275,29 @@ function startSyncWithRetry(watcherArray, maximumNumberOfAttempts, pauseSeconds)
 }
 
 app.listen(config.net.port, () => {
-  startSyncWithRetry(watchers, config.start.attempts, config.start.pauseSeconds);
-
-  debug('finder ' + config.version.major + '.' + config.version.minor + '.' +
-    config.version.bug + ' with api version ' + config.version.api +
-    ' waiting for requests on port ' + config.net.port);
+  //delete existing index and create new index with spatial mapping 
+  esclient.indices.exists({ index: config.elasticsearch.index })
+  .then(function (resp) {
+    if (resp) { //if config.elasticsearch.index already exists
+      //delete existing index
+      return esclient.indices.delete({ index: config.elasticsearch.index });
+    }
+  }).then(function (resp) {
+    //(re)create index
+    return esclient.indices.create({ index: config.elasticsearch.index });
+  }).then(function (resp) {
+    //add spatial mapping
+    return esclient.indices.putMapping({ index: config.elasticsearch.index, type: "compendia", body: mapping });
```

Comment:
Each call to `elasticsarch.indices` should have a `debug(..)` statement with its result.